stumpy/core.py

```diff
@@ -2559,3 +2573,183 @@ def _select_P_ABBA_value(P_ABBA, k, custom_func=None):
             MPdist = partition[k]
 
     return MPdist
+
+
+@njit(parallel=True)
+def _merge_topk_PI(PA, PB, IA, IB):
+    """
+    Merge two top-k matrix profiles PA and PB, and update PA (in place) while
+    always choosing values of PA over values of PB in case of ties. Also, update
+    IA accordingly.
+
+    Unlike `_merge_topk_ρI`, where `top-k` largest values are kept, this function
+    keeps `top-k` smallest values.
+
+    `PA` and `PB` are 2D arrays, with each row sorted ascendingly. To update `PA[i]`,
```

Comment:
> What is it that you want to convey that isn't clear?

I agree. It's too much.... I just provide it here for the sake of consistency as I provide a similar description for `_merge_topk_ρI`. 

I think `_merge_topk_ρI` is not easy to understand without such description. And, since I provide a description there, I provide a similar one here. I am going to erase this description here in function `_merge_topk_PI`; however, I will keep the description in  `_merge_topk_ρI` for now. Later, when you get a chance to take a look at it, you can let me know if it helps in that case or not.